### PR TITLE
feature: adding 'LspDiagThis'

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
// jump to the diag msg in this/current line
// start from current column of current line

### CODE REVIEW

**Comment by Shane-XB-Qian** [comment]
this was for pr# 162
i felt i still need this.

**Comment by yegappan** [comment]
> 
> Isn't this exactly how `LspDiagNext` and `LspDiagPrev` currently works?
>

Yes. I thought ":LspDiagNext" works like ":cnext" and jumps to the line with the next diagnostic. I forgot that it works like ":cafter" and jumps to the next nearest diagnostic.

File: autoload/lsp/diag.vim [hunk 1]
```diff
@@ -350,11 +350,12 @@ export def LspDiagsJump(lspserver: dict<any>, which: string): void
...
   var curlnum: number = line('.')
   var curcol: number = charcol('.')
-  for diag in (which == 'next') ? diags : diags->copy()->reverse()
+  for diag in (which == 'next' || which == 'this') ? diags : diags->copy()->reverse()
     var lnum = diag.range.start.line + 1
     var col = diag.range.start.character + 1
-    if (which == 'next' && (lnum > curlnum ||  lnum == curlnum && col > curcol))
+    if (which == 'next' && (lnum > curlnum || lnum == curlnum && col > curcol))
 	  || (which == 'prev' && (lnum < curlnum || lnum == curlnum && col < curcol))
+	  || (which == 'this' && (lnum == curlnum && col > curcol))
```
Comment (andlrc) [inline autoload/lsp/diag.vim:358]: I think that you want `col >= curcol` otherwise it will write the warnmsg if your cursor is on the diagnostic.

**Comment by Shane-XB-Qian** [comment]
This is just for 1 line, current line, i think simply press `0` is enough if/when you needed to turn around the search. 
No need to make things complex.

**Comment by Shane-XB-Qian** [comment]
and as you suggested at #162 at begining, `this` (or `here`) is just for jump, not display.
otherwise what's the different between `LspDiagCurrent` (or what i was modified for LspDiagCurrent) ?

**Comment by Shane-XB-Qian** [comment]
or whatever, let's just make diaghere be able to keep jumping the line.
i raised pr #183 wish you can merge.

**Pull request merged**